script.js

```diff
@@ -0,0 +1,81 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+// main function to generate the random password
+const getPasswordLength = () => {
+  /*
+   * All your code goes here to generate a password
+   */
+  // get the password length
+  const passwordLength = prompt(
+    "Enter a password length of 8 or more and 128 characters or less"
+  );
+  //storing the answer from prompt into passwordLength then getting the string and converting it to a number stored in passwordLengthNum and then returning it.
+  const passwordLengthNum = parseInt(passwordLength, 10);
+  //verifying the condition
+  if (passwordLengthNum >= 8 && passwordLengthNum <= 128) {
+    return passwordLengthNum;
+  } else {
+    alert("please enter a valid password length with valid criteria");
+  }
+};
+
+//Password criteria function
+const passwordCriteria = () => {
+  //make user to confirm their criteria
+  const lowercase = confirm("Would you like to include lowercase?");
+  //validating each confirm
+  const uppercase = confirm("Would you like to include uppercase?");
+  const numbers = confirm("Would you like to include numbers?");
+  const special = confirm("Would you like to include spacial characters?");
+  //array which is empty
+  const getCriteria = [];
+  if (lowercase) {
+    getCriteria.push("abcdefghijklmnopqrstuvwxyz");
```

Comment:
Pls consider declaring these values at the top of your code as constants the same for special characters, uppercase and the rest of them that is a better practice. Don't directly push the raw values  to the array